tests/integration/sync/test_sync_command.py

```diff
@@ -0,0 +1,398 @@
+    @classmethod
+    def setUpClass(cls):
+        cls.docker_client = docker.from_env()
+        cls.local_images = [
+            ("public.ecr.aws/sam/emulation-python3.7", "latest"),
+        ]
+        # setup some images locally by pulling them.
+        for repo, tag in cls.local_images:
+            cls.docker_client.api.pull(repository=repo, tag=tag)
+            cls.docker_client.api.tag(f"{repo}:{tag}", "emulation-python3.7", tag="latest")
+            cls.docker_client.api.tag(f"{repo}:{tag}", "emulation-python3.7-2", tag="latest")
+
+        # setup signing profile arn & name
+        cls.signing_profile_name = os.environ.get("AWS_SIGNING_PROFILE_NAME")
+        cls.signing_profile_version_arn = os.environ.get("AWS_SIGNING_PROFILE_VERSION_ARN")
+
+        PackageIntegBase.setUpClass()
+
+        cls.test_data_path = Path(__file__).resolve().parents[1].joinpath("testdata", "sync")
+
+        # To be used for code sync integ tests
+        # template_path = str(cls.test_data_path.joinpath("template.yaml"))
+        # cls.shared_stack_name = "accelerate-shared-stack"
+        # LOG.info(cls.shared_stack_name)
+
+        # command = "samdev" if os.getenv("SAM_CLI_DEV") else "sam"
+        # sync_command_list = [command, "sync", "-t", template_path, "--dependency-layer", "--stack-name", cls.shared_stack_name]
+        # run_command_with_input(sync_command_list, "y\n".encode())
+
+    def setUp(self):
+        self.cfn_client = boto3.client("cloudformation")
+        self.ecr_client = boto3.client("ecr")
+        self.lambda_client = boto3.client("lambda")
+        self.api_client = boto3.client("apigateway")
+        self.sfn_client = boto3.client("stepfunctions")
+        self.sns_arn = os.environ.get("AWS_SNS")
+        self.stacks = []
+        time.sleep(CFN_SLEEP)
+        super().setUp()
+
+    def tearDown(self):
+        shutil.rmtree(os.path.join(os.getcwd(), ".aws-sam", "build"), ignore_errors=True)
+        for stack in self.stacks:
+            # because of the termination protection, do not delete aws-sam-cli-managed-default stack
+            stack_name = stack["name"]
+            if stack_name != SAM_CLI_STACK_NAME:
+                region = stack.get("region")
+                cfn_client = (
+                    self.cfn_client if not region else boto3.client("cloudformation", config=Config(region_name=region))
+                )
+                ecr_client = self.ecr_client if not region else boto3.client("ecr", config=Config(region_name=region))
+                self._delete_companion_stack(cfn_client, ecr_client, self._stack_name_to_companion_stack(stack_name))
+                cfn_client.delete_stack(StackName=stack_name)
+        super().tearDown()
+
+    # To be used for code sync integ tests
+    # @pytest.fixture
+    # def clean_up_shared_stack(self):
+    #     pass
+
+    @parameterized.expand(["template.yaml"])
+    def test_sync_infra(self, template_file):
```

Comment:
Do we want to test updating the stack with infra sync as well?

Reply:
Yes, right now calling infra twice will cause many unexpected errors. I already have two sets of templates set up for updating tests. Once I figure out how to avoid the errors I will add it (probably before EOD).